pkgs/development/python-modules/torchvision/default.nix

```diff
@@ -34,24 +52,29 @@ in buildPythonPackage rec {
   };
 
   nativeBuildInputs = [ libpng ninja which ]
-    ++ lib.optionals cudaSupport [ cudatoolkit_joined ];
-
-  TORCHVISION_INCLUDE = "${libjpeg_turbo.dev}/include/";
-  TORCHVISION_LIBRARY = "${libjpeg_turbo}/lib/";
+    ++ lib.optional cudaSupport cuda-native-redist;
 
   buildInputs = [ libjpeg_turbo libpng ]
-    ++ lib.optionals cudaSupport [ cudnn ];
+    ++ lib.optional cudaSupport cuda-redist;
```

Comment:
please prefer `optionals`. See https://discourse.nixos.org/t/lib-optional-considered-harmful/7081 for more info. The fact that `optional` exists at all is a bit unfortunate.